src/engine.rs

```diff
@@ -1759,8 +1767,10 @@ mod tests {
         let (start, _) = engine.file_span(LogQueue::Append);
         engine.purge_expired_files().unwrap();
         assert!(start < engine.file_span(LogQueue::Append).0);
+        let expired_file_count = engine.expired_file_count(LogQueue::Append);
```

Comment:
Remove the `expired_file_count` trait method. We should only care about active file here. Disable recycling if it's needed.